gtsfm/frontend/detector_descriptor/detector_descriptor_base.py

```diff
@@ -43,9 +43,26 @@ def detect_and_describe(self, image: Image) -> Tuple[Keypoints, np.ndarray]:
             Corr. descriptors, of shape (N, D) where D is the dimension of each descriptor.
         """
 
+    def filter_by_response(self, keypoints: Keypoints, descriptors: np.ndarray) -> Tuple[Keypoints, np.ndarray]:
+        """Filter features according to their responses."""
```

Comment:
also both these methods are missing args and returns, if you still need them.

Reply:
The variable names are self-explanatory IMHO.

Reply:
The inputs, yes. but the return types could do with some clarification.

You are right, it makes perfect sense to us who use GtSfM regularly, but having the details (like shapes, lengths) in docstring helps new users I guess

Reply:
I think the shapes of input and return args would be useful to new users

Reply:
I think the return statement is also very easy to read with respect to understanding the outputs: `return keypoints.extract_indices(valid_idxs), descriptors[valid_idxs]`

I would pretty much just be copying and pasting the explanation from the docstring of `detect_and_describe` which is two lines above this function. I can do that if you feel strongly about a novice user not being able to understand the two lines of code.

Reply:
Let's go ahead and add the full docstring. I think almost always a good idea.

Reply:
Added